src/transformers/utils/quantization_config.py

```diff
@@ -1282,7 +1282,19 @@ def __init__(
                 sparsity_config.get("format"), **sparsity_config
             )
 
-        super().__init__(quant_method=QuantizationMethod.COMPRESSED_TENSORS)
+        self.quant_method = QuantizationMethod.COMPRESSED_TENSORS
+
+    def post_init(self):
+        if self.run_compressed:
+            if self.is_sparsification_compressed:
+                logger.warn(
+                    "`run_compressed` is only supported for quantized_compressed models"
+                    " and not for sparsified models. Setting `run_compressed=False`"
+                )
+                self.run_compressed = False
+            elif self.is_quantized and not self.is_quantization_compressed:
```

Comment:
Why do we have to check both conditions? Why not just is_qusntization_compressed?

Reply:
Yes this for cases, when the model is quantized but uncompressed

Reply:
Good catch! updated

Reply:
Good catch! Updated